clustermesh/clustermesh.go

```diff
@@ -510,6 +517,19 @@ func (k *K8sClusterMesh) GetClusterConfig(ctx context.Context) error {
 	return nil
 }
 
+func (k *K8sClusterMesh) getRunningCiliumVersion(ctx context.Context) error {
+	pods, err := k.client.ListPods(ctx, k.params.Namespace, metav1.ListOptions{LabelSelector: "k8s-app=cilium"})
+	if err != nil {
+		return fmt.Errorf("unable to list pods: %w", err)
+	}
+	for _, pod := range pods.Items {
+		image := pod.Spec.Containers[0].Image
+		version := strings.Split(image, ":")
+		k.imageVersion = version[1]
```

Comment:
I don't remember seeing this before, but FYI image tags can be of the form `docker.io/cilium/operator-generic:v1.9.7@sha256:8f121bf312654de62eeec0076755411b088ce4c48843b174d206ea1b141fbb7c` which has two colons, so the [1] index isn't exactly correct for this case (in fact, it will be wrong because it doesn't include the full sha).

Reply:
I believe `strings.SplitN("string", ":", 2)` solve this issue as image:tag syntax set in stone.

Reply:
Yep, that should work great. Thanks! :+1: